bug: Android builds targeting Java 1.8 which is not yet supported

The following commit upgraded the docker container to use Java 1.8:
https://github.com/karalabe/xgo/commit/52c3aeac29dc7f5a02ac0daaf818ba347af6b963

This has resulted in the Android .AAR files being built to target Java 1.8.  Trying to then build an Android application that depends on an .AAR produced after commit `52c3ae` results in the following because no current version of Android supports 1.8*:

```
:MyApplication:preDexDebug
Unknown source file : UNEXPECTED TOP-LEVEL EXCEPTION:
Unknown source file : java.lang.RuntimeException: Exception parsing classes
Unknown source file : 	at com.android.dx.command.dexer.Main.processClass(Main.java:752)
Unknown source file : 	at com.android.dx.command.dexer.Main.processFileBytes(Main.java:718)
Unknown source file : 	at com.android.dx.command.dexer.Main.access$1200(Main.java:85)
Unknown source file : 	at com.android.dx.command.dexer.Main$FileBytesConsumer.processFileBytes(Main.java:1645)
Unknown source file : 	at com.android.dx.cf.direct.ClassPathOpener.processArchive(ClassPathOpener.java:284)
Unknown source file : 	at com.android.dx.cf.direct.ClassPathOpener.processOne(ClassPathOpener.java:166)
Unknown source file : 	at com.android.dx.cf.direct.ClassPathOpener.process(ClassPathOpener.java:144)
Unknown source file : 	at com.android.dx.command.dexer.Main.processOne(Main.java:672)
Unknown source file : 	at com.android.dx.command.dexer.Main.processAllFiles(Main.java:574)
Unknown source file : 	at com.android.dx.command.dexer.Main.runMonoDex(Main.java:311)
Unknown source file : 	at com.android.dx.command.dexer.Main.run(Main.java:277)
Unknown source file : 	at com.android.dx.command.dexer.Main.main(Main.java:245)
Unknown source file : 	at com.android.dx.command.Main.main(Main.java:106)
Unknown source file : Caused by: com.android.dx.cf.iface.ParseException: bad class file magic (cafebabe) or version (0034.0000)
Unknown source file : 	at com.android.dx.cf.direct.DirectClassFile.parse0(DirectClassFile.java:472)
Unknown source file : 	at com.android.dx.cf.direct.DirectClassFile.parse(DirectClassFile.java:406)
Unknown source file : 	at com.android.dx.cf.direct.DirectClassFile.parseToInterfacesIfNecessary(DirectClassFile.java:388)
Unknown source file : 	at com.android.dx.cf.direct.DirectClassFile.getMagic(DirectClassFile.java:251)
Unknown source file : 	at com.android.dx.command.dexer.Main.parseClass(Main.java:764)
Unknown source file : 	at com.android.dx.command.dexer.Main.access$1500(Main.java:85)
Unknown source file : 	at com.android.dx.command.dexer.Main$ClassParserTask.call(Main.java:1684)
Unknown source file : 	at com.android.dx.command.dexer.Main.processClass(Main.java:749)
Unknown source file : 	... 12 more
Unknown source file : 1 error; aborting

:MyApplication:preDexDebug FAILED
```

References:
http://android-developers.blogspot.com.au/2016/03/first-preview-of-android-n-developer.html

--- a/docker/base/build.sh
+++ b/docker/base/build.sh
@@ -289,7 +289,7 @@ for TARGET in $TARGETS; do
 
         # Compile the Java wrapper and assemble into a .jar file
         mkdir -p $jni/build
-        javac -cp . -d $jni/build $jni/${package//.//}/*.java
+        javac -target 1.7 -source 1.7 -cp . -d $jni/build $jni/${package//.//}/*.java
         (cd $jni/build && jar cvf $archive/classes.jar *)
 
         # Finally assemble the archive contents into an .aar and clean up

